src/packs/checker.rs

```diff
@@ -258,28 +238,37 @@ impl<'a> CheckAllBuilder<'a> {
     fn build_strict_mode_violations(
         &mut self,
         recorded_violations: &'a HashSet<ViolationIdentifier>,
-    ) -> Vec<&'a ViolationIdentifier> {
-        let mut indexed_checkers: HashMap<
+    ) -> anyhow::Result<Vec<&'a ViolationIdentifier>> {
+        let indexed_checkers: HashMap<
             String,
             &Box<dyn CheckerInterface + Send + Sync>,
-        > = HashMap::new();
-        for checker in &self.found_violations.checkers {
-            indexed_checkers.insert(checker.violation_type(), checker);
-        }
+        > = self
+            .found_violations
+            .checkers
+            .iter()
+            .map(|checker| (checker.violation_type(), checker))
+            .collect();
 
-        let strict_mode_violations: Vec<&'a ViolationIdentifier> =
-            recorded_violations
-                .iter()
-                .filter(|v| {
-                    indexed_checkers
-                        .get(&v.violation_type)
-                        .unwrap()
-                        .is_strict_mode_violation(v, self.configuration)
-                })
-                .collect();
-        strict_mode_violations
+        recorded_violations
+            .iter()
+            .try_fold(vec![], |mut acc, violation| {
```

Comment:
I spent far too long trying to do this with `filter_map` and `transpose`. Then I discovered `try_fold`!